base/file.jl

```diff
@@ -554,8 +554,41 @@ end
 
 const temp_prefix = "jl_"
 
-if Sys.iswindows()
+# Use `Libc.rand()` to generate random strings
+function _rand_filename(len = 10)
+    slug = Base.StringVector(len)
+    chars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
```

Comment:
```suggestion
    chars = b"0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
```
You really want an array of bytes here, I think.

If `chars` is a string, then `chars[i]` converts the byte to a `Char` and then it gets converted back for assignment to `slug[i]`, and `length(chars)` is slow.

Reply:
Thanks @stevengj!

Reply:
This is a pretty decent speedup.  The old version:

```
julia> @btime _rand_filename(10)
  320.220 ns (2 allocations: 96 bytes)
```

The new version:
```
julia> @btime _rand_filename(10)
  66.308 ns (2 allocations: 96 bytes)
```